.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,37 @@
+name: Pre-commit
+permissions: read-all
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
+  merge_group:
+    branches: [main]
+
+jobs:
+  pre-commit:
+    name: Pre-commit
+    runs-on: ubuntu-latest
+    permissions:
+      # Give the default GITHUB_TOKEN write permission to commit and push the
+      # added or changed files to the repository.
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 7
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: pnpm
+      - run: pnpm --color install
+      - uses: pre-commit/action@v3.0.0
+      - uses: stefanzweifel/git-auto-commit-action@v4
```

Comment:
Does this work for forks?

Reply:
Good question. I'll give it a whirl

Reply:
Haven't tested yet, but looks like a bit of a hornet's nest and potentially introduces security vulnerability to get it to work 😬 https://github.com/stefanzweifel/git-auto-commit-action/issues/211

So do we switch to pre-commit lite and deal with really slow commit, or switch to the setup from my other attempt with syncing versions. Downside of that is we can't easily run custom fixers like if we wanted to autoformat our recorded tests using our custom yaml writer

Reply:
Hmm. It might be possible to get something working with the completion of this workflow triggering another workflow that runs in the `main` context? That'd be super complex though, and sounds like even more of a nightmare.

I've seen workflows that use a personal access token to get this sort of thing to work, but that's probably just as risky.